File: 프로그래머스/lv2/42888. 오픈채팅방/오픈채팅방.java
import java.util.*;

public class Solution {
	public String[] solution(String[] record) {

		List<String> List = new ArrayList<String>();
		Map<String, String> idmap = new HashMap<>();
		
		for(String str : record) {
			StringTokenizer token = new StringTokenizer(str);
			String cmd = token.nextToken();
			if(cmd.equals("Enter")||cmd.equals("Change")) {
				String id = token.nextToken();
				String name = token.nextToken();
				idmap.put(id, name);
			}
		}

		for(String str : record) {
			StringTokenizer token = new StringTokenizer(str);
			String cmd = token.nextToken();
			if(cmd.equals("Enter")) {
				String id = token.nextToken();
				List.add(idmap.get(id)+"님이 들어왔습니다.");
				
			}else if(cmd.equals("Leave")) {
				String id = token.nextToken();
				List.add(idmap.get(id)+"님이 나갔습니다.");
				
			}
		}
		
		String[] answer = new String[List.size()];
		List.toArray(answer);
		
		
		
		return answer;
	}
}

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.swift
import Foundation

func solution(_ record: [String]) -> [String] {
        var id_name: [String: String] = [:] //id : name
        var id_type: [(id: String, type: String)] = []
        var (type, id, name) = ("", "", "")
        
        var ans = [String]()
        for i in record {
            let input = i.split { $0 == " " }.map { String($0) }
            if input.count > 2 {
                // Enter, Change
                (type, id, name) = (input[0], input[1], input[2])
                id_name[id] = name
            } else {
                // Leave
                (type, id) = (input[0], input[1])
            }
            if type != "Change" {
                id_type.append((id: id, type: type))
            }
            

        }

        for i in 0 ..< id_type.count {
            let name = id_name[id_type[i].id]!
            let ment = id_type[i].type == "Enter" ? "님이 들어왔습니다." : "님이 나갔습니다."
            ans.append(name + ment)
        }
        return ans
    }